runtime/Java/src/org/antlr/v4/runtime/UnbufferedCharStream.java

```diff
@@ -82,7 +82,7 @@ public UnbufferedCharStream() {
 	/** Useful for subclasses that pull char from other than this.input. */
 	public UnbufferedCharStream(int bufferSize) {
 		n = 0;
-		data = new char[bufferSize];
+		data = new int[bufferSize];
 	}
 
 	public UnbufferedCharStream(InputStream input) {
```

Comment:
It seems like we should add an optional charset arg on one or more of the arguments. Also, doesn't the fill() method assume UTF-X? I.e., it would not work with some other encoding, right?